sos/report/plugins/openshift.py

```diff
@@ -82,6 +89,18 @@ def _check_oc_logged_in(self):
 
         # Not logged in currently, attempt to do so
         token = self.get_option('token') or os.getenv('SOSOCPTOKEN', None)
+
+        # Check first if we can use the localhost.kubeconfig before
+        # using token. We don't need to use 'host' option due we
+        # use localhost from kubeconfig
+        if self._check_localhost_kubeconfig():
+            self.exec_cmd(env={'KUBECONFIG': self.master_localhost_kubeconfig})
+
+            oc_res = self.exec_cmd("oc login -u system:admin \
+                --insecure-skip-tls-verify=True")
```

Comment:
We talked a little about this over GChat so I'm going to rehash a bit of that now so we can get some wider input. However, one note is that I should have been more specific here - I meant adding the `env` parameter to the `self.exec_cmd()` call for `oc login`, e.g.:

```python
self.exec_cmd(
    'oc login -u system:admin --insecure-skip-tls-verify=True',
    env={'KUBECONFIG': self.master_localhost_kubeconfig}
)
```

That still brings us around to the issue of needing to have the `KUBECONFIG` env var set for any subsequent `oc` commands.

Generally speaking there are only 2 solutions at the top level here:

1. Set the env var on the system
2. Pass the env var with each `add_cmd_output()` command.

Option 1 is a non-starter for me, as we have to maintain the guarantee of "no system changes". I understand that in this specific context and use case that is not as large of a concern as it is normally, but I don't want to set a precedent (or a precedent for an exception), especially since we have historically always rejected directly setting env vars in sos in the past.

Option 2 gives us 3 routes. Either we add `--kubeconfig=$path` to each command and then have to do a lot of tinkering with `suggest_filename`, which is also a non-starter in my opinion. Or we pass `env={'KUBECONFIG': self.master_localhost_kubeconfig}` to every `add_cmd_output()` call, which is _very_ ugly and error prone, especially for future maintenance.

I then had a 3rd idea of allowing `Plugin` to set a "default environment" configuration, which would get automatically passed via `env` to every `add_cmd_output()` call. Sort of similar to how we have `Plugin.set_predicate()` today.

I think this is a workable solution and overall a Good Thing(tm) for plugins, but I'd like to hear if there are any counter-arguments or better ideas first.

To be clear, my thought is something like this:


```python

class MyPlugin(Plugin):

    def setup(self):
        self.set_default_environment({'KUBECONFIG': self.master_localhost_kubeconfig})
        self.add_cmd_output('oc something') 
```
And in `Plugin.add_cmd_output()` we add logic to pass that dict to `sos_get_command_output()` via the `env` parameter.

Reply:
Hey @TurboTurtle, thx for summarizing our gchat discussion. I totally agree with your 3rd idea. Having a env variable for the sos report would be the best way to do that. I agree, options 1 and 2 are not feasible.

Reply:
I've posted #2894 which should fit this usecase. If you want to try that out with a working cluster ontop of this branch, you'd do something like this:

```python
if self._check_localhost_kubeconfig():
            self.set_default_cmd_environment({'KUBECONFIG': self.master_localhost_kubeconfig})

            oc_res = self.exec_cmd("oc login -u system:admin \
                --insecure-skip-tls-verify=True")
```

Reply:
Hey @TurboTurtle , yeah, I think it fits perfectly to our use case, having the default variable defined. I will check it on my lab.